plugin/remotecall/main.go

```diff
@@ -0,0 +1,108 @@
+//go:build full || e2e
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tg123/sshpiper/libplugin"
+	"github.com/tg123/sshpiper/plugin/internal/remotecall"
+	"github.com/urfave/cli/v2"
+)
+
+func createRemoteCaller(c *cli.Context) (*remotecall.RemoteCall, error) {
+	remoteCall, err := remotecall.InitRemoteCall(
+		c.String(userClusterEndpoint),
+		c.Generic(remoteEndpoints).(*remotecall.StringMapFlag).Value,
+		c.Path(mappingKeyPath),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating remote caller: %w", err)
+	}
+
+	return remoteCall, nil
+}
+
+func getUserName(response *remotecall.UserKeyAuthResponse) string {
```

Comment:
maybe call it `generateUpstreamUserName` or sth like that to avoid confusion between incoming and outgoing user naem